tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go

```diff
@@ -0,0 +1,136 @@
+			ibiImageSetName string
+		)
+		BeforeAll(func() {
+			if MGMTConfig.Cluster == nil {
+				Skip("Failed to collect cluster info")
+			}
+
+			if MGMTConfig.SeedClusterInfo == nil || MGMTConfig.SeedClusterInfo.SeedClusterOCPVersion == "" {
+				Skip("Seed clusterinfo not supplied")
+			}
+
+			splitOCPVersion := strings.Split(MGMTConfig.SeedClusterInfo.SeedClusterOCPVersion, ".")
+
+			if len(splitOCPVersion) < 2 {
+				Skip("Could not determine short OCP version from seed cluster ocp version")
+			}
+
+			ibiImageSetName = strings.Join(splitOCPVersion[:2], ".")
+		})
+
+		It("is successful in connected environment", func() {
+
+			tsparams.ReporterNamespacesToDump[MGMTConfig.Cluster.Info.ClusterName] = "spoke namespace"
+
+			By("Create namespace for IBI installation")
+			_, err := namespace.NewBuilder(APIClient, MGMTConfig.Cluster.Info.ClusterName).Create()
+			Expect(err).NotTo(HaveOccurred(), "error creating namespace")
+
+			By("Get pull secret from cluster")
+			spokePullSecret, err := secret.Pull(APIClient, "pull-secret", "openshift-config")
+			Expect(err).NotTo(HaveOccurred(), "error getting pull-secret from hub cluster")
+
+			By("Create pull secret for spoke cluster")
+			spokePullSecret.Definition.Name = MGMTConfig.Cluster.Info.ClusterName
+			spokePullSecret.Definition.Namespace = MGMTConfig.Cluster.Info.ClusterName
+			spokePullSecret.Definition.ResourceVersion = ""
+			_, err = spokePullSecret.Create()
+			Expect(err).NotTo(HaveOccurred(), "error creating spoke pull-secret")
+
+			for host, info := range MGMTConfig.Cluster.Info.Hosts {
+				By("Create baremetalhost secret for " + host)
+				_, err = secret.NewBuilder(
+					APIClient, host, MGMTConfig.Cluster.Info.ClusterName, v1.SecretTypeOpaque).WithData(map[string][]byte{
+					"username": []byte(info.BMC.User),
+					"password": []byte(info.BMC.Password),
+				}).Create()
+				Expect(err).NotTo(HaveOccurred(), "error creating bmh secret")
+
+				By("Create baremetalhost for " + host)
+				hostBMH := bmh.NewBuilder(
+					APIClient, host, MGMTConfig.Cluster.Info.ClusterName, info.BMC.URLv4, host, info.BMC.MACAddress, "UEFI")
+				hostBMH.Definition.Spec.AutomatedCleaningMode = "disabled"
+
+				_, err = hostBMH.Create()
+				Expect(err).NotTo(HaveOccurred(), "error creating baremetalhost")
+			}
+
+			By("Create cluster deployment for IBI installation")
+			_, err = hive.NewABMClusterDeploymentBuilder(
+				APIClient, MGMTConfig.Cluster.Info.ClusterName,
+				MGMTConfig.Cluster.Info.ClusterName, MGMTConfig.Cluster.Info.ClusterName,
+				MGMTConfig.Cluster.Info.BaseDomain, MGMTConfig.Cluster.Info.ClusterName, metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"dummy": "label",
+					},
+				}).WithPullSecret(MGMTConfig.Cluster.Info.ClusterName).Create()
+			Expect(err).NotTo(HaveOccurred(), "error creating cluster deployment")
+
+			var snoNodeName string
+
+			for hostname := range MGMTConfig.Cluster.Info.Hosts {
+				snoNodeName = hostname
+
+				break
```

Comment:
Hmm, do we have to use loop (and break) here?

Reply:
It's just an easy way to get the SNO node's name until we support multinode